misk-crypto/src/main/kotlin/misk/crypto/CryptoModule.kt

```diff
@@ -79,52 +102,137 @@ class CryptoModule(
  * Extension function for convenient encryption of [ByteString]s.
  * This function also makes sure that no extra copies of the plaintext data are kept in memory.
  */
+@Deprecated(
+    message = "Prefer to use encryptin methods that accept a Map<String, String> for encryption context",
+    replaceWith = ReplaceWith(
+        expression = "aead.encrypt(plaintext, encryptionContext)",
+        imports = ["misk.crypto.CryptoModule.encrypt"]))
 fun Aead.encrypt(plaintext: ByteString, aad: ByteArray? = null): ByteString {
   val plaintextBytes = plaintext.toByteArray()
   val encrypted = this.encrypt(plaintextBytes, aad)
   plaintextBytes.fill(0)
   return encrypted.toByteString()
 }
 
+/**
+ * Extension function for convenient encryption of [ByteString]s
+ * using a map of strings as encryption context.
+ * This function also makes sure that no extra copies of the plaintext data are kept in memory.
+ */
+fun Aead.encrypt(
+  plaintext: ByteString,
+  encryptionContext: Map<String, String>? = null
+) : ByteString {
+  val serializedEncryptionContext = encryptionContext?.let { serialize(it) }
+  val plaintextBytes = plaintext.toByteArray()
+  val encryptedByteString = this.encrypt(plaintextBytes, serializedEncryptionContext)
+  plaintextBytes.fill(0)
+  return CiphertextFormatter.serialize(encryptedByteString, encryptionContext)
+}
+
 /**
  * Extension function for convenient decryption of [ByteString]s.
  * This function also makes sure that no extra copies of the plaintext data are kept in memory.
  */
+@Deprecated(
+    message = "Prefer to use encryption methods that accept a map for encryption context",
+    replaceWith = ReplaceWith(
+        expression = "aead.decrypt(plaintext, encryptionContext)",
+        imports = ["misk.crypto.CryptoModule.decrypt"]))
 fun Aead.decrypt(ciphertext: ByteString, aad: ByteArray? = null): ByteString {
   val decryptedBytes = this.decrypt(ciphertext.toByteArray(), aad)
   val decrypted = decryptedBytes.toByteString()
   decryptedBytes.fill(0)
   return decrypted
 }
 
+/**
+ * Extension function for convenient decryption of [ByteString]s
+ * using a map of strings as encryption context.
+ * This function also makes sure that no extra copies of the plaintext data are kept in memory.
+ */
+fun Aead.decrypt(
+  ciphertext: ByteString,
+  encryptionContext: Map<String, String>? = null
+) : ByteString {
+  val ciphertextByteString = CiphertextFormatter.deserialize(ciphertext, encryptionContext)
```

Comment:
What should happen if we call `Aead.decrypt` on a ciphertext that was created with a version of misk-crypto before this PR was merged?